src/main/java/org/isf/users/rest/UserController.java

```diff
@@ -43,215 +45,221 @@
-     */
-    @ResponseStatus(HttpStatus.CREATED)
-    @PostMapping(value = "/users", produces = MediaType.APPLICATION_JSON_VALUE)
-    public UserDTO newUser(@Valid @RequestBody UserDTO userDTO) throws OHServiceException {
-        LOGGER.info("Attempting to create user {}.", userDTO.getUserName());
-        User user = userMapper.map2Model(userDTO);
-        try {
-            LOGGER.info("User successfully created.");
-            return userMapper.map2DTO(userManager.newUser(user));
-        } catch (OHServiceException serviceException) {
-            LOGGER.info("User is not created.");
-            throw new OHAPIException(new OHExceptionMessage("User not created."));
-        }
-    }
+	/**
+	 * Creates a new {@link User}.
+	 * @param userDTO - the {@link User} to insert
+	 * @return the new created user.
+	 * @throws OHServiceException When failed to create user
+	 */
+	@ResponseStatus(HttpStatus.CREATED)
+	@PostMapping(value = "/users", produces = MediaType.APPLICATION_JSON_VALUE)
+	public UserDTO newUser(@Valid @RequestBody UserDTO userDTO) throws OHServiceException {
+		LOGGER.info("Attempting to create user {}.", userDTO.getUserName());
+		User user = userMapper.map2Model(userDTO);
+		try {
+			LOGGER.info("User successfully created.");
+			return userMapper.map2DTO(userManager.newUser(user));
+		} catch (OHServiceException serviceException) {
+			LOGGER.info("User is not created.");
+			throw new OHAPIException(new OHExceptionMessage("User not created."));
+		}
+	}
 
-    /**
-     * Deletes an existing {@link User}.
-     *
-     * @param username - the name of the {@link User} to delete
-     */
-    @ResponseStatus(HttpStatus.NO_CONTENT)
-    @DeleteMapping(value = "/users/{username}", produces = MediaType.APPLICATION_JSON_VALUE)
-    public void deleteUser(@PathVariable String username) throws OHServiceException {
-        User foundUser = userManager.getUserByName(username);
-        if (foundUser == null) {
-            throw new OHAPIException(new OHExceptionMessage("User not found."));
-        }
-        try {
-            userManager.deleteUser(foundUser);
-        } catch (OHServiceException serviceException) {
-            throw new OHAPIException(new OHExceptionMessage("User not deleted."));
-        }
-    }
+	/**
+	 * Deletes an existing {@link User}.
+	 * @param username - the name of the {@link User} to delete
+	 */
+	@ResponseStatus(HttpStatus.NO_CONTENT)
+	@DeleteMapping(value = "/users/{username}", produces = MediaType.APPLICATION_JSON_VALUE)
+	public void deleteUser(@PathVariable String username) throws OHServiceException {
+		User foundUser = userManager.getUserByName(username);
+		if (foundUser == null) {
+			throw new OHAPIException(new OHExceptionMessage("User not found."), HttpStatus.NOT_FOUND);
+		}
+		try {
+			try {
+				userManager.deleteUser(foundUser);
+			} catch (OHServiceException ex) {
+				foundUser.setDeleted(true);
+				userManager.deleteUser(foundUser);
+			}
```

Comment:
should API not doing it automatically but just report with a error and the UI send a DTO already set for Deleted=True?

Reply:
Up to two request ? It could add an unncessary complexity level and it's error prone. By doing that, you should also send an explicit code to let the frontend know that it's a containt violation error. Does it really worth ?

Reply:
I think that anyway you have to inform the client about what is happening and why a user is hard-deleted and another might be only soft-delete. If I think_ about this, I easily end up with having the same logic as GUI

Reply:
Since on the ui we could not rely on the error messsage due to the fact that it is not yet standardized, we could notify the user before the deletion that the target object will be just marked as deleted if he's used somewhere. What do you think ?

Reply:
but add a comment to improve that logic when the errors will be standardized